src/adapter/src/coord/command_handler.rs

```diff
@@ -363,7 +363,15 @@ impl Coordinator {
              mock_nonce: String,
              nonce: String,
              tx: oneshot::Sender<Result<SASLChallengeResponse, AdapterError>>| {
-                let opts = mz_auth::hash::mock_sasl_challenge(&role_name, &mock_nonce);
+                let opts = mz_auth::hash::mock_sasl_challenge(
+                    &role_name,
+                    &mock_nonce,
+                    &Some(
+                        self.catalog()
+                            .system_config()
+                            .default_password_hash_iterations(),
+                    ),
+                );
```

Comment:
Rather than the system variable, shouldn't it extract it from the number of iterations stored in the role's hashed password? i.e. if the hased password is `SCRAM-SHA-256$<iterations>:<salt>$<stored_key>:<server_key>`, it'd extract it from `iterations`?

Reply:
> Rather than the system variable, shouldn't it extract it from the number of iterations stored in the role's hashed password?

Nope! The mock challenge is used when the role does not exist. 

> As an extension, would really like to see a testdrive test along the lines of...

That test makes sense, the nature of how scram's challenge is stored/issued is always that the iterations are stored and passed along to the thing validating (in password auth us, in scram the client). I'll ensure we have such a test, if not in testdrive than in the auth module.